### ISSUE TITLE
The pointer is not initalized while receiving indices of Sift keypoints

### ISSUE THREAD

**Issue by yashmotorai** [PointCloudLibrary/pcl#5281]
Hi Guys,
After extracting Sift keypoints. I am trying to get the indices of the key points for further work(to find the normal of specific indexes).  I am trying to do something like this
std::vector<int> indices = sift.getKeypointsIndices() ->indices. I am getting the following error. 

/usr/include/boost/smart_ptr/shared_ptr.hpp:734: typename boost::detail::sp_member_access<T>::type boost::shared_ptr<T>::operator->() const [with T = const pcl::PointIndices; typename boost::detail::sp_member_access<T>::type = const pcl::PointIndices*]: Assertion `px != 0' failed.

This error is related to pointers, I think while initializing ConstPtr in the source code the initialization is missing with pcl::PointIndices. Something like this 

pcl::PointIndices keypoint_indices;
pcl::PointIndices::ConstPtr keypoints_indices_cptr(&keypoint_indices);

Can you check it once, if what I am doing is incorrect, or is there something missing in the source code?

**Comment by mvieth** [comment]
This indeed seems to be unimplemented for SIFT. I found a pull request that tries to add this: https://github.com/PointCloudLibrary/pcl/pull/1648
However, there are merge conflicts and some not-yet-addressed requested changes. I will take a closer look later.

**Comment by mvieth** [comment]
I thought about it and it is correct that this is unimplemented for SIFT. The keypoints returned by SIFT are _new_ points, _not_  a subset of points from the input cloud (as some other keypoint algorithms return). This is because SIFT downsamples the initial cloud several times. I checked the implementation of the pull request I linked above, and the indices returned there basically belong to random points in the input cloud. So `getKeypointsIndices` makes no sense for SIFT.
